src/StreamingDataFrames/README.md

```diff
@@ -1,3 +1,258 @@
+easy by managing an entire infrastructure stack (Kafka, Kubernetes, CI/CD) for you 
+under the hood, with production software best practices in mind, and all backed by Git. 
+
+[Be sure to check us out at quix.io](https://quix.io/) to get up and running with 
+Kafka, fast!
+
+<br>
+
+## Upcoming Improvements
+
+- Stateful Windowing: tumbling, hopping, and sliding windows
+- Group-bys and joins (for merging topics/keys)
+- State recovery based on Kafka changelog topics
+- Support for exactly-once Kafka processing (aka transactions)
+- Other serialization support like Avro and Protobuf support
+- Schema Registry support
+- Multiple input topic support (for non-stateful applications)
+
+<br>
+
+## Want to get Involved?
+
+We strongly believe in the open source model and would love to build a strong 
+community with you, so please join us!
+
+While we are still working on out contribution guidelines, here are some ways you can 
+get involved right now:
+
+- [Join our community mailing list](https://share.hsforms.com/1WWc91mC_SqCvk8xeUWibMw4yjw2)
+- [Check out our project on GitHub](https://github.com/quixio/quix-streams/)
+- [Hang out with us on Slack](https://join.slack.com/t/stream-processing/shared_invite/zt-25lo3wxo7-28hlegDhR2XqyLWjBQynwA)
+
+
+<br><br>
+
+# The Basics
+
+Here's the minimum you need to know to get started with Streaming DataFrames.
+
+More detailed breakdowns can be found in the [documentation folder](./documentation).
+
+## The `Application` class
+
+This is the handler and entrypoint of any `quixstreams` application.
+
+The `Application` class houses all kafka-related configuration, and can generate all 
+necessary objects you will need to get up and running.
+
+At minimum, you'll need to know your Kafka broker address and the consumer group ID you
+wish to use.
+
+You can get started via:
+```python
+from quixstreams import Application
+
+app = Application(
+   broker_address="my_broker_url",
+   consumer_group="my_consumer_group_name",
+)
+```
+
+<br>
+
+### Using the Quix Platform? - `Application.Quix()`
+
+If you are using the Quix platform, you will still use the `Application` class, but
+instead call it via `Application.Quix()`.
+
+If you are running this within the Quix platform it will be configured 
+automatically. Otherwise, see 
+[**Quix Platform Configuration**](./documentation/quix_platform.md).
+
+<br>
+
+## Defining Kafka Topics: `Application.topic()`
+
+Any topic you plan to interact with will require a respective `Topic` object. 
+
+To generate a topic, simply call `Application.topic(name)`, like so:
+
+```
+input_topic = app.topic("my_input_topic")
+output_topic = app.topic("my_output_topic")
+```
+
+Hold on to those topic objects, you'll need them later for your `dataframe`!
+
+<br>
+
+## Creating your DataFrame: `Application.dataframe()`
+
+Next is the `StreamingDataFrame`, which the `Application` class can generate for you 
```

Comment:
I understand that dataframe term is common in Python and developers should know what it means.
But I know I wasn't familiar with it in the beginning so I would maybe add a short sentence explaining dataframe concept. Here or at some other place